server/src/main/java/io/crate/expression/scalar/SubscriptFunction.java

```diff
@@ -245,8 +245,9 @@ public Query toQuery(Function parent, Function inner, Context context) {
         // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
         //       inner
         if (!(inner.arguments().get(0) instanceof Reference ref
-                && inner.arguments().get(1) instanceof Literal<?>
-                && parent.arguments().get(1) instanceof Literal<?> cmpLiteral)) {
+              && inner.arguments().get(1) instanceof Literal<?>
+              && parent.arguments().size() == 2 // parent could for example be `op_isnull`
```

Comment:
just double checking: is there any case that we need to check for `>= 2` instead of  `== 2`?

Reply:
thx!